### PR TITLE
Enhancment/workspace and sidesheet configuration

### PR DESCRIPTION
**Pull request by Noggling**
# Description

Please include a summary of the change and which issue is fixed.
List any external dependencies that are required for this change.

Completes: [AB#214138](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/214138) [AB#214142](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/214142)

## Checklist:

-   [x] I have performed a self-review of my own code.
-   [x] I have linked my DevOps task using the AB# tag.
-   [x] My code is easy to read, and comments are added where needed.
-   [ ] My code is covered by tests.

### CODE REVIEW

File: src/Core/Client/Framwork/index.ts [hunk 1]
```diff
@@ -24,13 +24,13 @@ export const initClient = async (init: Configurator): Promise<Client> => {
...
     };
-    window.Client = client;
+    window.lighhouse = client;
     return client;
 };
 
 declare global {
     interface Window {
-        Client: Client;
+        lighhouse: Client;
```
Comment (Noggling) [inline src/Core/Client/Framwork/index.ts:33]: lighthouse

File: src/Core/WorkSpace/Readme.md [hunk 2]
```diff
@@ -1,4 +1,73 @@
...
+    app: {
+        appType: 'Workspace',
+        setup: scopeChangeSetup,
+        SidesheetComponent: SidesheetWrapper as any,
+    },
+    tags: [],
+    appEnv: 'prod',
+}
+
+const workspaceManifest = {
```
Comment (Noggling) [inline src/Core/WorkSpace/Readme.md:27]: Oppdater

File: src/Core/WorkSpace/src/Components/DataViewerHeader/Header.tsx [hunk 3]
```diff
@@ -8,6 +7,7 @@ import { isProduction } from '@equinor/lighthouse-portal-client';
 import { StatusBar } from '@equinor/lighthouse-status-bar';
 import { useMemo } from 'react';
 import { FilterFilled } from '../../../../../components/Icon/FilterIconFilled';
+import { useDataCreator } from '../../../../../FusionModules/DataCreatorReact/Hooks/useCreator';
```
Comment (Noggling) [inline src/Core/WorkSpace/src/Components/DataViewerHeader/Header.tsx:10]: fix alias

File: src/Core/WorkSpace/src/WorkSpaceApi/Types/WorkspaceConfig.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { SidesheetApi } from '@equinor/sidesheet';
+import { ResolverFunction } from './functions';
+import { SidesheetComponentManifest, SidesheetWidgetManifest } from './SideSheets';
+
+export interface WorkSpaceSideSheet<T, SideSheetIds extends string>
```
Comment (Noggling) [inline src/Core/WorkSpace/src/WorkSpaceApi/Types/WorkspaceConfig.ts:5]: sjekk om dette er i bruk

File: src/Core/WorkSpace/src/WorkSpaceApi/WorkSpaceTypes.ts [hunk 5]
```diff
@@ -22,19 +21,19 @@ export interface DataSource<T> {
...
     objectIdentifier: keyof T;
     viewerId: string;
     defaultTab?: WorkspaceTab;
-    dataFactoryCreator(factory: Factory): void;
-    openSidesheet(SidesheetContent?: React.FC<any>, props?: any, appName?: string): void;
+    openSidesheet: OpenSidesheetFunc;
     CustomSidesheet?: React.FC<{ item: T; actions: SidesheetApi }>;
+    customSidesheetOptions?: WorkSpaceSideSheet<T, SideSheetId>;
+    customGroupeSidesheet?: WorkSpaceSideSheet<any, string>;
+    CustomGroupeSidesheet?: React.FC<{ item: any; actions: SidesheetApi }>;
```
Comment (Noggling) [inline src/Core/WorkSpace/src/WorkSpaceApi/WorkSpaceTypes.ts:36]: remove CustomGroupeSidesheet | CustomSidesheetList | CustomSidesheet

File: src/FusionModules/DataCreatorModule/configurator.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    constructor() {
+        this.configuration = {} as DataCreatorConfig;
+    }
+
+    configure(configuration: DataCreatorConfig): void {
+        this.configuration = configuration;
+    }
+}
+
+export function setupCreator({
```
Comment (Noggling) [inline src/FusionModules/DataCreatorModule/configurator.ts:27]: maybe Remove

File: src/FusionModules/DataCreatorModule/configurator.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+            parentApp: props.parentApp,
+        },
+    };
+
+    return <T extends CreatorType>(type: T) =>
+        ({
+            CreatorManifest: creatorManifest,
+            CreatorComponent: {
+                widgetId,
+                widgetType,
```
Comment (Noggling) [inline src/FusionModules/DataCreatorModule/configurator.ts:57]: sett widgetType

File: src/apps/DisciplineReleaseControl/DisciplineReleaseControlWidgets.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+});
+
+export const rcSidesheetCreator = setupWorkspaceSidesheet<Pipetest, 'rc'>({
+    id: 'rc',
+    color: '#0084C4',
+    component: ReleaseControlSidesheet,
+    props: {
+        objectIdentifier: 'name',
+        parentApp: 'piping-and-ht',
+        function: (id: string) => {
```
Comment (Noggling) [inline src/apps/DisciplineReleaseControl/DisciplineReleaseControlWidgets.ts:26]: fix

File: src/apps/DisciplineReleaseControl/DisciplineReleaseControlWidgets.ts [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+import { HTSidesheet, Pipetest } from './Types/pipetest';
+
+export const htSidesheetCreator = setupWorkspaceSidesheet<HTSidesheet, 'ht'>({
+    id: 'ht',
+    color: '#0084C4',
+    component: ReleaseControlHTSidesheet,
+    props: {
+        objectIdentifier: 'value',
+        parentApp: 'piping-and-ht',
+        function: (id: string) => {
```
Comment (Noggling) [inline src/apps/DisciplineReleaseControl/DisciplineReleaseControlWidgets.ts:13]: fix

File: src/apps/DisciplineReleaseControl/WorkspaceConfig/DataCreator/dataCreatorConfig.tsx [hunk 10]
```diff
@@ -1,11 +1,19 @@
...
+    widgetId: 'htCreator',
+    widgetType: 'creator',
     title: 'Create release control workflow',
-    accessCheck: () => Promise.resolve(false),
-    onClick: () => {
-        openSidesheet(DisciplineReleaseControlFactoryComponent, undefined, 'piping-and-ht');
+    color: '#7B3A96',
+    widget: DisciplineReleaseControlFactoryComponent,
+    props: {
+        accessCheckFunctionId: 'htCreatorAccess',
```
Comment (Noggling) [inline src/apps/DisciplineReleaseControl/WorkspaceConfig/DataCreator/dataCreatorConfig.tsx:11]: move to relese controll

**Review by Gustav-Eikaas — APPROVED** [review]

**Pull request merged**